mini-project/BillManagementSystem/accounting/schema/schema.py

```diff
@@ -0,0 +1,207 @@
+                        user_id: Optional[int] = None, 
+                        title: Optional[str] = None, 
+                        category_id: Optional[int] = None) -> Bill:
+    
+    if not any((id, user_id, title, category_id)):
+        raise ValueError("Hey! Please query by at least one parameter!")
+    
+    if id is not None:
+        bill = models.Bill.objects.get(id=id)
+        return [bill]
+    else:
+        queryset = models.Bill.objects.all()
+        
+        if user_id:
+            user = models.User.objects.get(id=user_id)
+            bill = queryset.filter(user=user)
+            
+        if title:
+            bill = queryset.filter(title__icontains=title)
+            
+        if category_id:
+            category = models.Category.objects.get(id=category_id)
+            bill = queryset.filter(category=category)
+            
+        return bill
+
+@strawberry.type
+class Query:
+    category: CategoryResponse = strawberry.field(resolver=get_category_by_id)
+    categories: List[Category] = strawberry.field(resolver=get_all_categories)
+    user: List[User] = strawberry.field(resolver=get_all_users)
+    bill: List[Bill] = strawberry.field(resolver=get_bills_by_filter)
+
+@strawberry.type
+class Mutation:
+    @strawberry.mutation
+    def add_user(self, register_input: RegisterInput) -> AuthResponse:
+        django_user = AdminUser.objects.create_user(
+            username=register_input.username, 
+            email=register_input.email, 
+            password=register_input.password, 
+            first_name=register_input.first_name, 
+            last_name = register_input.last_name
+        )
+
+        accounting_user = models.User(user=django_user, birthday=register_input.birthday, phone_number=register_input.phone_number)
+        accounting_user.save()
+        
+        new_user = User(
+            id=accounting_user.id,
+            username=django_user.username,
+            first_name=django_user.first_name,
+            last_name=django_user.last_name,
+            email=django_user.email,
+            birthday=accounting_user.birthday,
+            phone_number=accounting_user.phone_number,
+            bills= None,
+            individual_spendings=None
+        )
+        token = default_token_generator.make_token(django_user)
+        
+        return AuthResponse(success=True, token=token, user=new_user)
+    
+    @strawberry.mutation
+    def add_category(self, category: CategoryInput) -> CategoryResponse:
+        category = models.Category(name=category.name)
+        try:
+            category.save()
+            return CategoryResponse(success=True, category=category)
+        except IntegrityError as e:
+            return CategoryResponse(success=False, category=None, error="This category is taken. Don't create similar category!") 
+        except ValidationError as e:
+            return CategoryResponse(success=False, category=None, error=str(e))
```

Comment:
Do leave these kinds of try catch to strawberry instead of catching them yourself